pkg/cluster/sync.go

```diff
@@ -1594,6 +1619,27 @@ func (c *Cluster) syncLogicalBackupJob() error {
 			if reason != "" {
 				c.logger.Infof("reason: %s", reason)
 			}
+			if strings.Contains(reason, "annotations do not match") {
+				annotationToRemoveTemplate := `{"spec":{"jobTemplate":{"spec":{"template":{"metadata":{"annotations":{`
+				for anno := range job.Spec.JobTemplate.Spec.Template.Annotations {
+					if _, ok := desiredJob.Spec.JobTemplate.Spec.Template.Annotations[anno]; !ok {
+						// template annotation was removed
+						for _, ignore := range c.OpConfig.IgnoredAnnotations {
+							if anno == ignore {
+								continue
+							}
+						}
+						annotationToRemoveTemplate += fmt.Sprintf(`"%s":null,`, anno)
+					}
+				}
+				annotationToRemoveTemplate = strings.TrimSuffix(annotationToRemoveTemplate, ",") + `}}}}}}}`
+				job, err = c.KubeClient.CronJobs(c.Namespace).Patch(context.TODO(),
+					jobName, types.StrategicMergePatchType, []byte(annotationToRemoveTemplate), metav1.PatchOptions{}, "")
+				if err != nil {
+					c.logger.Errorf("failed to remove annotations from the logical backup job %q pod template: %v", jobName, err)
+					return err
+				}
+			}
```

Comment:
This part is still using the string concat patch contruction ... 

Again more patch code is following down the line. Is there no other was to do it. The code is becoming harder to understand with the extra comparison and patch code. It looks redundant with minor differences.

Reply:
i have forgotten refactor this part regarding the marshalling